modules/user/user.service.js

```diff
@@ -229,34 +224,51 @@ class UserService {
     return savedUser;
   }
 
+  async sendConfirmationLetter(email, language) {
+    const user = await this.getUserByFieldOrThrow('email', email);
+    const token = await generateToken(user._id, user.email);
+    const message = {
+      from: process.env.MAIL_USER,
+      to: user.email,
+      subject: 'Confirm Email',
+      html: confirmationMessage(user.firstName, token, language),
+    };
+    await sendEmail(message);
+    return true;
+  }
+
   async deleteUser(id) {
     const res = await User.findByIdAndDelete(id);
     return res || new Error(USER_NOT_FOUND);
   }
 
   async confirmUser(token) {
     const decoded = jwt.verify(token, process.env.SECRET);
-    const user = await this.getUserByFieldOrThrow('email', decoded.email);
-    user.confirmed = true;
-    await User.findByIdAndUpdate(user._id, user);
-    return true;
+    return await User.findByIdAndUpdate(
+      decoded.userId,
+      { confirmed: true },
+      { new: true },
+    );
   }
 
   async recoverUser(email, language) {
     const user = await User.findOne({ email });
     if (!user) {
       throw new UserInputError(USER_NOT_FOUND, { statusCode: 404 });
     }
+
     const token = await generateToken(user._id, user.email, {
       EXPIRES_IN: process.env.RECOVERY_EXPIRE,
     });
+    user.recoveryToken = token;
     const message = {
       from: process.env.MAIL_USER,
       to: email,
       subject: 'Recovery Instructions',
```

Comment:
Please, rename subject to `[HORONDI] Instructions for recovery of password`

Reply:
renamed